asserts/fetcher.go

```diff
@@ -60,6 +63,54 @@ func NewFetcher(trustedDB RODatabase, retrieve func(*Ref) (Assertion, error), sa
 	}
 }
 
+type SequenceFormingFetcher interface {
+	// SequenceFormingFetcher must also implement the interface of the Fetcher.
+	Fetcher
+
+	// FetchSequence retrieves the assertion as indicated by its sequence reference.
+	FetchSequence(*AtSequence) error
+}
+
+// NewSequenceFormingFetcher creates a SequenceFormingFetcher which will use trustedDB to
+// fetch sequence forming assertions using retrieveSeq, prerequisites assertions using retrieve,
+// and then will pass them to save, saving prerequisites before the depending assertions.
```

Comment:
// NewSequenceFormingFetcher creates a SequenceFormingFetcher which will use trustedDB to determine trusted assertions,
// will fetch assertions following prerequisites using retrieve and sequence-forming assertions using retrieveSeq, and then will pass
// them to save, saving prerequisites before dependent assertions.

Reply:
Thanks for the clarifications made here, fixed.